lib/puppet/file_serving/metadata.rb

```diff
@@ -88,6 +90,21 @@ def collect_stat(path)
   def collect(source_permissions = nil)
     real_path = full_path
 
+    if static_catalog
+      if environment && environment.static_catalogs?
+        env_path = File.join(Puppet[:environmentpath], environment.name.to_s) # environment.name is a symbol sometimes
+        puts "Checking if #{real_path} is in #{env_path}"
+        if real_path.start_with? env_path
```

Comment:
> so we can throw a compilation error on the latter

I would be reluctant to throw a compilation error, because we don't do that now. For example, a file resource can have multiple sources, and it's ok if some of the them don't exist on the master.